src/doc/contrib/book.toml

```diff
@@ -4,3 +4,6 @@ authors = ["Eric Huss"]
 
 [output.html]
 git-repository-url = "https://github.com/rust-lang/cargo/tree/master/src/doc/contrib/src"
+
+[output.html.redirect]
+"/apidoc/cargo/" = "https://doc.rust-lang.org/nightly/nightly-rustc/cargo/"
```

Comment:
That's my fault. The redirect need to point to a file instead of a directory. Otherwise mdbook refuses to build (see this [action failure](https://github.com/weihanglo/cargo/actions/runs/3902991307/jobs/6667280160) in my own fork).

```suggestion
"/apidoc/cargo/index.html" = "https://doc.rust-lang.org/nightly/nightly-rustc/cargo/"
```

Besides that, it would be great if we can redirect each page under `/apidoc/cargo/*` to the new correct location, or lead it to at least the root of where the new doc lives. Could you help with this? It's fine that we cannot make it, assumed that nightly cargo doc is not really widely used.

Reply:
Sure I can do that. But the way I'd do this is build the api docs locally, run a script on it and add a redirect for each file to the `redirect` table, like:

```toml
[output.html.redirect]
"/apidoc/cargo/index.html" = "https://doc.rust-lang.org/nightly/nightly-rustc/cargo/index.html"
"/apidoc/cargo/all.html" = "https://doc.rust-lang.org/nightly/nightly-rustc/cargo/all.html"
...
```

this will probably be a long list, is this how you'd expect the solution to look like?

Reply:
Sorry not being clear. I was hoping to avoid that kind of change. Instead, by the best effort we might find a simple way to redirect them all, such as a general 404 page for GitHub page. The whole https://doc.crates.io is in fact served as a GH page.

<img width="355" alt="image" src="https://user-images.githubusercontent.com/14314532/212104247-44707dcb-4937-4433-82e9-5ac9f47e35d4.png">


If you find it too complicated, that means it is not worth your time. We can then move on :)

Reply:
Hmm, maybe we can add a simple `/apidoc/cargo/404.html` file that looks something like:

```html
<html>
  <head>
    <meta http-equiv="refresh" content="0; url=https://doc.rust-lang.org/nightly/nightly-rustc/cargo/" />
  </head>
  <body></body>
</html>
``` 

This way we would redirect only after a 404 is given. I think I'll tinker around with this for a bit, see if I can make this work with gh-pages.

Reply:
Ok, here my observations:

1. using a custom `404.html` file with gh-pages only works if it resides in the root directory (so only `doc.crates.io/404.html` will change the default 404 page)
2. there is the unused [`contrib/404.html`](https://github.com/rust-lang/cargo/blob/gh-pages/contrib/404.html) file on the gh-pages branch. Not important, just a side note :smile: 
3. the cargo book has been moved in the past from `https://doc.crates.io/` to `https://doc.rust-lang.org/cargo/`. There are still some remaining files that redirect to the new location in the [`gh-pages` branch](https://github.com/rust-lang/cargo/tree/gh-pages). I have no idea how the cargo book was moved, but maybe we can use the same approach as was used back then? Just for the sake of consistency I think this would be best. 

I don't know how the redirecting files on the `gh-pages` branch were created (the history is wiped with every commit from the `Deploy Contrib` workflow). If this were a personal repository of mine, I probably would just merge this PR (to disable the creation of the files in the `apidoc/` folder), delete all non-html files and replace all the html files in `apidoc/` with a redirection to the new destination. But I don't know about your policies about committing directly to the `gh-pages` branch.